### ISSUE TITLE
Make IntervalQuery available via the Query DSL

### ISSUE THREAD

**Issue by romseygeek** [elastic/elasticsearch#29636]
Lucene 7.4 adds a new IntervalQuery, intended as a longer-term replacement for the Spans family.  This should be available through the Query DSL.

The query looks something like this:
```
POST _search
{
  "query": {
    "intervals": {
      "my_text": {
        "all_of": {
          "max_gaps": 1,
          "ordered": true,
          "intervals": [
            {
              "match": {
                "query": "cold slimy porridge",
                "analyzer": "",
                "ordered": false,
                "max_gaps": 1
              }
            },
            {
              "match": {
                "query": "cold slimy porridge",
                "analyzer": "",
                "ordered": false,
                "max_gaps": 1
              }
            }
          ],
          "filter": {
            "containing": {
              "match": {
                "query": "cold slimy porridge",
                "analyzer": "",
                "ordered": false,
                "max_gaps": 1
              }
            }
          }
        }
      }
    }
  }
}
```

**Comment by mattweber** [comment]
@romseygeek Can you please make interval sources pluggable while working on this?

**Comment by romseygeek** [comment]
Closed by #36135
